### ISSUE TITLE
`lambda: (yield)` causes spurious errors

### ISSUE THREAD

**Issue by ambv** [python/mypy#3551]
Consider:
```py3
noop_ctxmgr = contextmanager(lambda: (yield))
```
Error:
```
f.py:11: error: Cannot infer type argument 1 of "contextmanager"
f.py:11: error: Yield value expected
```

This passes with no errors:
```py3
@contextmanager
def noop_ctxmgr():
    yield
```

**Comment by ilevkivskyi** [comment]
It looks like this is specific to lambda expressions. Things like this work:
```python
def noop_ctxmgr() -> Iterator[None]:
    yield

test = contextmanager(noop_ctxmgr)
```

Also if there is a yield value, the first error is still present:
```python
noop_ctxmgr = contextmanager(lambda: (yield 42))
# E: Cannot infer type argument 1 of "contextmanager"
```
